src/traces/splom/base_plot.js

```diff
@@ -168,46 +167,40 @@ function makeGridData(gd) {
     return gridBatches;
 }
 
-function clean(newFullData, newFullLayout, oldFullData, oldFullLayout, oldCalcdata) {
-    var oldModules = oldFullLayout._modules || [];
-    var newModules = newFullLayout._modules || [];
+function clean(newFullData, newFullLayout, oldFullData, oldFullLayout) {
+    oldLoop:
+    for(var i = 0; i < oldFullData.length; i++) {
+        var oldTrace = oldFullData[i];
 
-    var hadSplom, hasSplom;
-    var i;
+        if(oldTrace.type === 'splom') {
+            for(var j = 0; j < newFullData.length; j++) {
+                var newTrace = newFullData[j];
 
-    for(i = 0; i < oldModules.length; i++) {
-        if(oldModules[i].name === 'splom') {
-            hadSplom = true;
-            break;
-        }
-    }
-    for(i = 0; i < newModules.length; i++) {
-        if(newModules[i].name === 'splom') {
-            hasSplom = true;
-            break;
-        }
-    }
+                if(oldTrace.uid === newTrace.uid && newTrace.type === 'splom') {
+                    continue oldLoop;
+                }
+            }
 
-    if(hadSplom && !hasSplom) {
-        for(i = 0; i < oldCalcdata.length; i++) {
-            var cd0 = oldCalcdata[i][0];
-            var trace = cd0.trace;
-            var scene = cd0.t._scene;
-
-            if(
-                trace.type === 'splom' &&
-                scene && scene.matrix && scene.matrix.destroy
-            ) {
-                scene.matrix.destroy();
-                cd0.t._scene = null;
+            if(oldFullLayout._splomScenes) {
+                var scene = oldFullLayout._splomScenes[oldTrace.uid];
+                if(scene && scene.destroy) scene.destroy();
+                // must first set scene to null in order to get garbage collected
```

Comment:
interesting... how did you figure this out?

Reply:
Checking GPU memory consumption in the Chrome task manager (`<shift>+<esc>`) before and after calling `Plotly.purge(gd)`.

Reply:
Alright, I have to wonder
- How general that observation is - does this apply across browsers/OS's? does it apply when the console is closed?
- Assuming it does apply in some real-world situations, where else should we be doing the same thing? Would it be worthwhile making a `Lib.delete(container, key)` or something to encourage reuse?